### ISSUE TITLE
problem starting the SW on windows 10

### ISSUE THREAD

**Issue by pmario** [sukima/tw-bookcase#2]
Hi, 
I did open power-shell and installed as follows:

```
npm install -g tw-bookcase
d:
cd \tmp
tw-bookcase --version 
```

and got

```
Usage: tw-bookcase CONFIG COMMAND [ARGS]
```

**Comment by pmario** [comment]
The same thing happens with every command.

**Comment by sukima** [comment]
Confirmed. It is a typo I think. Will do research and post a fix soon. Thank you @pmario

**Comment by sukima** [comment]
I just did a bit of research and have not found where the problem exists. I followed the industry standard for defining a global executable and I tested it on both Linux and MacOS.

There has to be something to do with your version of NPM on Windows. Maybe there was a bug that has since been fixed in new versions of NPM?

At the moment I am unable to move forward with this as I am not a Windows user. I would not even know where to start on Windows. I am left to ask others for help.

### Research Links

* https://medium.com/netscape/a-guide-to-create-a-nodejs-command-line-package-c2166ad0452e
* https://stackoverflow.com/questions/25333570/npm-package-json-bin-wont-work-on-windows
* https://www.sitepoint.com/javascript-command-line-interface-cli-node-js/

**Comment by sukima** [comment]
@pmario I assume that TiddlyWiki itself works using the same install process? Maybe there is a way to compare my `package.json` with TiddlyWiki's `package.json`. Care to confirm that TW does work globally on Windows?

**Comment by pmario** [comment]
TiddlyWiki works.

**Comment by pmario** [comment]
I think the command is executed, since it shows the help message. ... but it seems to have a problem with the parameters.

**Comment by pmario** [comment]
I'm using node version 8 atm. ... Did you use stuff, that only works with node 10+?

**Comment by pmario** [comment]
I did update to node 10. ... now --version works, but ... 

```
tw-bookcase asdf --init
```

gives me this `tw-bookcase.info` file

```
{
    "wikis": []
}
```

I need to create the `asdf` directory otherwise I get an error message. 

Do I need to create a `tiddlywiki asdf --init server` first?

**Comment by pmario** [comment]
Hmm, ... It seems there is a [path.sep problem](https://nodejs.org/api/path.html#path_path_sep) in https://github.com/sukima/tw-bookcase/blob/2911df6bb9fe49661d6597d64eeb20606cc0b723/libs/tiddlywiki-instance.js#L42 ... and probably a lot more positions. 

Windows uses `\`

**Comment by pmario** [comment]
I do get this error message: 

```
$ tw-bookcase asdf/ --listen
Serving on http://127.0.0.1:C:\Users\Mario\AppData\Roaming\npm\node_modules\tw-bookcase\app\tw-bookcase-socket
(press ctrl-C to exit)
Warning: Plugins required for client-server operation ("tiddlywiki/filesystem" and "tiddlywiki/tiddlyweb") are missing from tiddlywiki.info                                file
events.js:174
```

So TW doesn't find the instance ...

**Comment by sukima** [comment]
Will investigate further. To answer your question, I use Node 10+ features.

**Comment by sukima** [comment]
> It seems there is a path.sep problem...

This part of the code is working with **URL paths** which will always be forward slashes. This is used to inject a config tiddler into the running TiddlyWiki instance so that its HTTP saver can route to the correct sub path.

> I need to create the `asdf` directory otherwise I get an error message.

This is a bug and I opened a new issue for it #3 

> Do I need to create a `tiddlywiki asdf --init server` first?

Yes and No. By default no as you can click the **create wiki** button in the GUI which will do this for you. But if you want to do it manually (or have already done it previously) you can use the **import wiki** button in the GUI instead.

CC @pmario

**Comment by pmario** [comment]
hmm, why did you close it. IMO it's not resolved.

**Comment by pmario** [comment]
as soon as I open localhost:8088/ i get an error as mentionrd in: https://github.com/sukima/tw-bookcase/issues/2#issuecomment-496986856

**Comment by pmario** [comment]
how should the directory structure look like?
are there any environment variables?

**Comment by bimlas** [comment]
> Will investigate further. To answer your question, I use Node 10+ features.

I know this is not your fault, but it is interesting that in package.json `"engines": { "node": ">=10" }` is set, but it does not indicated to the user on `npm install`.

https://docs.npmjs.com/files/package.json#engines

In any case, this should be clarified in readme.

**Comment by bimlas** [comment]
The readme says:

> `tw-bookcase mynewwiki --init` to create a folder for a new wiki that includes server-related components

@pmario said:

> I need to create the `asdf` directory otherwise I get an error message.

This message is displayed for me as well if the directory does not exist (using Windows + Node 10).

**Comment by sukima** [comment]
@pmario I moved the actual bug (bad error message) to issue #3 That left any items in **this** issue resolved and that is why I closed it.

Are you receiving some other issue? My Windows knowledge is so bad I could not even imagine trying to know how to fix a Windows only issue. I am *begging* for help from other more Windows friendly users to resolve Windows issues.

@pmario if this is different then the `--init` issue you described above (and in #3) let me know so I can reopen or make a new issue to track that.

@bimlas this is an NPM thing. Yarn does warn users but NPM just don't care. Tghere isn't much that can be done about that other then better documentation and/or use of a transpiler like Babel and I figured this project was so small and very likely not likely to be used much (I'm a small fry). But if it turns out people want Node 8 support I could look into transpiling down to ES5.
